patches/server/0809-Add-Material-getPlacedMaterial.patch

```diff
@@ -0,0 +1,47 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Jake Potrebic <jake.m.potrebic@gmail.com>
+Date: Wed, 29 Sep 2021 13:23:24 -0700
+Subject: [PATCH] Add Material#getPlacedMaterial
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java b/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java
+index 408863b137b7bda3f3e654ac3665ddeefb6d9e7b..6ca6e70788c0c230de6196266f1673af32f2ba57 100644
+--- a/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java
++++ b/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java
+@@ -118,6 +118,7 @@ public final class CraftMagicNumbers implements UnsafeValues {
+     // Paper start
+     private static final Map<org.bukkit.entity.EntityType, net.minecraft.world.entity.EntityType<?>> ENTITY_TYPE_ENTITY_TYPES = new HashMap<>();
+     private static final Map<net.minecraft.world.entity.EntityType<?>, org.bukkit.entity.EntityType> ENTITY_TYPES_ENTITY_TYPE = new HashMap<>();
++    private static final Map<Material, Material> PLACED_MATERIALS = new java.util.EnumMap<>(Material.class);
```

Comment:
never looked into this before: is an enum map still better than a normal hashmap on such a giant enum? could it possibly degrade to something worse? dont have time to do the research myself right now, maybe somebody knows, lol

Reply:
It's basically an array lookup vs a map lookup, for most cases the former is gonna be faster than the latter afaik